dev/io.openliberty.concurrent.internal/src/io/openliberty/concurrent/internal/processor/ContextServiceDefinitionBinding.java

```diff
@@ -198,17 +225,33 @@ static final String toString(ContextServiceDefinition anno) {
                         .append(", cleared=").append(Arrays.toString(anno.cleared())) //
                         .append(", propagated=").append(Arrays.toString(anno.propagated())) //
                         .append(", unchanged=").append(Arrays.toString(anno.unchanged())) //
+                        .append(", qualifiers=").append(Arrays.toString(anno.qualifiers())) //
                         .append(")");
         return b.toString();
     }
 
     @Trivial
-    private static final String toString(String[] list) {
+    private static final <T> String toString(T[] list) {
         if (list == null || list.length == 0)
             return "Unspecified";
         boolean none = true;
         for (int i = 0; none && i < list.length; i++)
-            none &= list[i] == null || list[i].length() == 0;
+            none &= list[i] == null || list[i].toString().isEmpty();
         return none ? "None" : Arrays.toString(list);
     }
+
+    @Trivial
+    private static final Class<?>[] toQualifierClassArray(String[] classList) throws IllegalArgumentException {
+        Class<?>[] clazzArray = new Class<?>[classList.length];
+        for (int i = 0; i < classList.length; i++) {
+            try {
+                //TODO is there a certain classloader I should be using to load this class? ApplicationClassLoader?
+                clazzArray[i] = Class.forName(classList[i]);
```

Comment:
Yes, we need to be careful with making it a requirement to load classes if we aren't certain we can have the right class loader when doing it.  There might be different class loaders between modules and for the application.  If we don't have access to the correct class loader here, another option would be to perform the merge as String class names, to be loaded later.  I'm not sure what is available.  You'll need to work with Tom on this part.